elixir/wordcount.ex

```diff
@@ -0,0 +1,29 @@
+IO.stream(:stdio, :line)
+# Create a dictionary in which the keys are the words and the values are
+# the number of appearances in the input text.
+|> Enum.reduce(%{}, fn (line, acc) ->
+  line
+  |> String.strip
+  |> String.split(~r/\s/)
+  |> Enum.reduce(acc, fn (word, counter) ->
+    if String.length(word) == 0 do
```

Comment:
Problem matching the empty string in a double headed anonymous function might be better readable and more efficient.